File: CRM-Server/app/schemas/customer_ai.py
"""AI 客户活动解析 Schema."""
from pydantic import BaseModel, Field
from typing import Optional


class CustomerAIParseRequest(BaseModel):
    """AI 解析客户活动请求"""
    content: str = Field(..., min_length=1, max_length=20000, description="用户输入的自然语言描述")
    customer_id: str = Field(..., description="客户对外 ID")
    customer_name: str = Field(..., description="客户名称")


class CustomerAICreateRequest(BaseModel):
    """AI 创建客户活动请求（用户确认后提交）"""
    customer_id: str = Field(..., description="客户对外 ID")
    customer_name: str = Field(..., description="客户名称")
    content: str = Field(..., description="活动原始内容")
    method: Optional[str] = Field(default=None, description="活动方式（电话/微信/拜访/邮件/线上会议/线下会议）")
    next_action: Optional[str] = Field(default=None, description="下一步动作")
    next_follow_time: Optional[str] = Field(default=None, description="下次跟进时间")

File: CRM-Client/src/api/customerAI.ts
/**
 * AI 客户活动解析 API
 */
import request from '@/utils/request'
import { z } from 'zod'

export interface CustomerAIParseRequest {
  content: string
  customer_id: number
  customer_name: string
}

export interface CustomerAIActivityInfo {
  content: string | null
  method: string | null
  next_action: string | null
  next_follow_time: string | null
}

export interface CustomerAIParseSSEEvent {
  event: 'status' | 'content' | 'parsed' | 'error'
  message?: string
  content?: string
  customer_id?: number
  customer_name?: string
  follow_up_info?: CustomerAIActivityInfo
}

export interface CustomerAICreateRequest {
  customer_id: number
  customer_name: string
  content: string
  method?: string | undefined
  next_action?: string | undefined
  next_follow_time?: string | undefined
}

const CustomerAICreateResponseSchema = z.object({
  id: z.number(),
  customer_id: z.number(),
  source_content: z.string(),
  activity_kind: z.string()
})

type CustomerAICreateResponse = z.infer<typeof CustomerAICreateResponseSchema>

export const customerAiApi = {
  /**
   * AI 解析客户活动信息（SSE 流式响应）
   */
  parseSSE: async (
    data: CustomerAIParseRequest,
    onEvent: (event: CustomerAIParseSSEEvent) => void,
    token: string
  ): Promise<void> => {
    const url = '/api/v1/customers/ai/parse'

    const response = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`
      },
      body: JSON.stringify(data)
    })

    if (!response.ok) {
      throw new Error(`HTTP error: ${response.status}`)
    }

    const reader = response.body?.getReader()
    if (!reader) {
      throw new Error('No response body')
    }

    const decoder = new TextDecoder()
    let buffer = ''

    while (true) {
      const { done, value } = await reader.read()
      if (done) break

      buffer += decoder.decode(value, { stream: true })
      const lines = buffer.split('\n\n')
      buffer = lines.pop() ?? ''

      for (const line of lines) {
        if (line.startsWith('data: ')) {
          try {
            const eventData = JSON.parse(line.slice(6)) as CustomerAIParseSSEEvent
            onEvent(eventData)

            if (eventData.event === 'parsed' || eventData.event === 'error') {
              return
            }
          } catch {
            // 忽略解析错误
          }
        }
      }
    }
  },

  /**
   * 从 AI 解析结果创建客户活动
   */
  create: async (data: CustomerAICreateRequest): Promise<CustomerAICreateResponse> => {
    // eslint-disable-next-line crmwolf/require-zod-schema
    const raw: unknown = await request.post('/v1/customers/ai/create', data)
    return CustomerAICreateResponseSchema.parse(raw)
  }
}
